Earthfile

```diff
@@ -240,14 +240,18 @@ COPY_SOURCE:
 
 rpm-build:
     FROM +init --base fedora:rawhide
-    RUN __install rpmdevtools gcc-c++ make openssl-devel unzip git ccache \
-                  findutils patch cmake doxygen "cmake(bson-1.0)" libdfp-devel
     GIT CLONE https://src.fedoraproject.org/rpms/libmongocrypt.git /R
+    # Install the packages listed by "BuildRequires" and rpm-build:
+    RUN __install $(awk '/^BuildRequires:/ { print $2 }' /R/libmongocrypt.spec) \
+                  rpm-build
     DO +COPY_SOURCE
-    WORKDIR /R
-    RUN cp -r /s/libmongocrypt/. .
-    RUN awk -f etc/rpm/tweak.awk < libmongocrypt.spec > libmongocrypt.2.spec
-    RUN rpmbuild -ba libmongocrypt.2.spec -v -D "_sourcedir $PWD"
+    RUN cp -r /s/libmongocrypt/. /R
+    RUN awk -f /R/etc/rpm/tweak.awk < /R/libmongocrypt.spec > /R/libmongocrypt.2.spec
+    RUN rpmbuild -ba /R/libmongocrypt.2.spec \
+        -D "_topdir /X" \
+        -D "_sourcedir /R"
+    SAVE ARTIFACT /X/RPMS /rpm/
+    SAVE ARTIFACT /X/SRPMS /rpm/
```

Comment:
I don't recognize the `SAVE` directive, so I'm assuming that it's related to Earthly rather than directly to Docker.  That said, I can't seem to find what, if anything, is done with these RPMs after they are built.  Note that in the C driver, which uses a script to configure a `mock` environment and perform the build within that, the last thing which we do is build a sample program to ensure that the system's toolchain are able to properly reference the installed resources:

```sudo mock -r ${config} --use-bootstrap-image --isolation=simple --cwd "/tmp/${build_dir}" --chroot -- /bin/sh -c "(
  rpm -Uvh ../mock-result/*.rpm &&
  gcc -I/usr/include/libmongoc-1.0 -I/usr/include/libbson-1.0 -o example-client src/libmongoc/examples/example-client.c -lmongoc-1.0 -lbson-1.0
  )"
```

Ideally, this task would install the resulting RPMs in the rawhide environment and then perhaps compile `./test/example-state-machine.c`.

I should have made a comment along these same in the PR (#554) where you added the Debian task to this file.

Reply:
Yes, `SAVE` is an Earthly thing that tells it was files to "export" from a build (by default they are thrown into the void).

I'll define a task to build a simple project.